elements/pl-units-input/pl-units-input.mustache

```diff
@@ -0,0 +1,186 @@
+    <span class="badge text-danger badge-invalid">
+        <i class="fa fa-exclamation-triangle" aria-hidden="true"></i> Invalid
+    </span>
+    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border"
+        data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true"
+        style="margin-left: 5px" title="Format Error" tabindex="0" data-content="{{parse_error}}">
+        Why
+        <i class="fa fa-question-circle" aria-hidden="true"></i>
+    </a>
+    {{/parse_error}}
+
+    {{#missing_input}}
+    {{#label}}<span>{{{label}}}</span>{{/label}}
+    <span class="badge text-dark badge-missing-input">
+        <i class="fa fa-question-circle" aria-hidden="true"></i> Missing Input
+    </span>
+    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border"
+        data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true"
+        style="margin-left: 5px" title="Format Error" tabindex="0"
+        data-content="There is no submitted value for this field.  This may have happened because the question was changed by course staff after the answer was submitted.">
+        Why
+        <i class="fa fa-question-circle" aria-hidden="true"></i>
+    </a>
+    {{/missing_input}}
+</span>
+{{/error}}
+
+{{^error}}
+<div class="d-inline-block">
+    {{#label}}<span>{{{label}}}</span>{{/label}}
+    <samp class="user-output">{{a_sub}}</samp>
+    {{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+
+    <!-- Show submitted answer submission was parsed from -->
+    <a href="javascript:void(0);" role="button" class="btn btn-sm btn-secondary small border"
+        data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true"
+        style="margin-left: 5px" title="Original Input" tabindex="0"
+        data-content="Parsed from <samp class=user-output>{{raw_submitted_answer}}</samp>">
+        <i class="fa fa-question-circle" aria-hidden="true"></i>
+    </a>
+</div>
+
+{{#feedback}}
+    <div class="d-inline-flex pl-units-input-feedback">
+        {{{feedback}}}
+    </div>
+{{/feedback}}
+
+{{#correct}}
+&nbsp;
+<span class="badge badge-success">
+    <i class="fa fa-check" aria-hidden="true"></i> 100%
+</span>
+{{/correct}}
+{{#partial}}
+&nbsp;
+<span class="badge badge-warning">
+    <i class="far fa-circle" aria-hidden="true"></i> {{partial}}%
+</span>
+{{/partial}}
+{{#incorrect}}
+&nbsp;
+<span class="badge badge-danger">
+    <i class="fa fa-times" aria-hidden="true"></i> 0%
+</span>
+{{/incorrect}}
+{{/error}}
+
+{{#inline}}</span>{{/inline}}
+{{/submission}}
+
+
+{{#answer}}
+{{#label}}<span>{{{label}}}</span>{{/label}}
+<samp>{{a_tru}}</samp>
+{{#suffix}}<span>{{suffix}}</span>{{/suffix}}
+{{/answer}}
+
+
+{{#format}}
+<!--
+    Might need separate format errors for first and last inputs.
+    TODO: more detailed format error messages
+-->
```

Comment:
Is this still a valid TODO?

Reply:
Nah, playing with it now the format errors are pretty detailed. Will remove. There's another TODO about more test cases, but that's handled by backend testing. Will remove that too.